xrpl/models/transactions/deposit_preauth.py

```diff
@@ -39,14 +45,87 @@ class DepositPreauth(Transaction):
         init=False,
     )
 
+    authorize_credentials: Optional[List[Credential]] = None
+    """The credential(s) that received the preauthorization. (Any account with these
+    credentials can send preauthorized payments)."""
+
+    unauthorize_credentials: Optional[List[Credential]] = None
+    """The credential(s) whose preauthorization should be revoked."""
+
     def _get_errors(self: Self) -> Dict[str, str]:
         errors = super()._get_errors()
         if self.authorize and self.unauthorize:
             errors["DepositPreauth"] = (
                 "One of authorize and unauthorize must be set, not both."
             )
 
-        if not self.authorize and not self.unauthorize:
-            errors["DepositPreauth"] = "One of authorize and unauthorize must be set."
+        if (
+            self.authorize is None
+            and self.unauthorize is None
+            and self.authorize_credentials is None
+            and self.unauthorize_credentials is None
+        ):
+            errors["DepositPreauth"] = (
+                "Exactly one field must be set for either authorize, unauthorize, "
+                + "authorize_credentials or unauthorize_credentials. It is "
+                + "invalid if none of the params are specified."
+            )
+
+        # Filter for input parameters which are provided in the transaction
+        # Optional fields in the transaction model are set to None
+        params = [
+            self.authorize,
+            self.unauthorize,
+            self.authorize_credentials,
+            self.unauthorize_credentials,
+        ]
+        if sum(param is not None for param in params) > 1:
+            errors["DepositPreauth"] = (
+                "DepositPreauth txn accepts exactly one input amongst `authorize`, "
+                + "`unauthorize`, `authorize_credentials` and `unauthorize_credentials`"
+                + "."
+            )
+
+        def _validate_credentials_length(
+            credentials: List[Credential], field_name: str
+        ) -> None:
+            if len(credentials) == 0:
```

Comment:
If you add a check like this at the top, you can remove the ifs below when you call this helper function
```
if credentials is None:
    return
````

Reply:
thanks, I've incorporated in cd7f8607